lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2799,6 +2800,9 @@ class HSAContext final : public KalmarContext
     */
     hsa_agent_t host;
 
+    cxlProfiler profiler;
+    cxlMarker contextMarker = CXL_MARKER_CLASS;
```

Comment:
@whchung we want to use this infrastructure to profile hcc RT itself.  The issue with putting it in mcwamp_hsa won't work well because then all the other libraries (i.e. libhc_am, libmcwamp) will  have to link against mcwamp_hsa.  
I think most of the code in hc_profile could be moved into a header; however, there's a class static  flag cxlMarker::enable that we need to put into a .cpp but it looks hcc doesn't have a .so that's being linked to by all the .so in hcc RT.   
We may need to create a new shared library for that.
If we build libhc_am, libmcwamp_hsa and friends and link them against hc_profile, then user won't need to worry about the new shared library or codexl's.